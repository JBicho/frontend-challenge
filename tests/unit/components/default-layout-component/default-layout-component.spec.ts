import DefaultLayout from '@/components/default-layout-component/default-layout-component.vue';
import { shallowMount } from '@vue/test-utils';
import Vue from 'vue';

describe('Default layout component tests', () => {
  let wrapper: any;

  test('Snapshot test', () => {
    const titles = {
      title: 'Test Title',
      subtitle: 'Test Subtitle'
    };

    wrapper = shallowMount(DefaultLayout, {
      propsData: { titles },
      stubs: ['router-link']
    });

    expect(wrapper).toMatchSnapshot();
  });

  test('Should display the values on props when mounted with props', () => {
    const titles = {
      title: 'Test Title',
      subtitle: 'Test Subtitle'
    };

    wrapper = shallowMount(DefaultLayout, {
      propsData: { titles },
      stubs: ['router-link']
    });

    expect(wrapper.find('.main-title').text()).toBe(titles.title);
    expect(wrapper.find('.subtitle').text()).toBe(titles.subtitle);
  });

  test('Should display default error messages when no props are passed', () => {
    const titles = {
      title: 'A Problem occurred',
      subtitle: 'Please refresh the page or try again in a bit'
    };

    wrapper = shallowMount(DefaultLayout, {
      propsData: {},
      stubs: ['router-link']
    });

    expect(wrapper.find('.main-title').text()).toBe(titles.title);
    expect(wrapper.find('.subtitle').text()).toBe(titles.subtitle);
  });
});
