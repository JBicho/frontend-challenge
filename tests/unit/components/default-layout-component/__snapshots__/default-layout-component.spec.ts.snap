// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default layout component tests Snapshot test 1`] = `
<main class="landing-page">
  <hgroup class="landing-page--titles">
    <h1 class="main-title">Test Title</h1>
    <h2 class="subtitle">Test Subtitle</h2>
  </hgroup>
  <div tabindex="0" class="landing-page--links">
    <router-link-stub title="Link to transcriptions page" tag="a" to="/transcription-list">Transcriptions</router-link-stub>
  </div>
</main>
`;
